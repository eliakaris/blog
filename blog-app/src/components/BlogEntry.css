article
{
    width: 720px;
}

article img
{
    display: block;
    padding: 5px;
    max-width: 720px
}