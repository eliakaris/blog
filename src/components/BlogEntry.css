article
{
    width: 720px;
    margin: auto;
}

article img
{
    display: block;
    padding: 5px;
    max-width: 720px;
    border: 1px solid #ccc;
}