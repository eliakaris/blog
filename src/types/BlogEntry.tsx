// src/types/BlogEntry.tsx

export interface BlogEntryData {
  pretty_pub_date: string;
  html: string;
  summary: string;
  slug: string;
  title: string;
  pub_date: string;
}

export interface BlogListEntry {
  slug: string;
  title: string;
  pub_date: string;
}