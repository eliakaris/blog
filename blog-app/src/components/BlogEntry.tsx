import * as React from 'react';
import * as Request from 'superagent';
import './BlogEntry.css';

export interface Props {
  match: {
    params: {
      slug: string;
    }
  };
}

interface BlogEntryData {
  pretty_pub_date: string;
  html: string;
  summary: string;
  slug: string;
  title: string;
  pub_date: string;
}

interface State {
  blogEntry: BlogEntryData;
}

class BlogEntry extends React.Component<Props, State> {

  componentWillMount() {
    if (this.props && this.props.match && this.props.match.params && this.props.match.params.slug) {
      Request.get(`/api/blog/${this.props.match.params.slug}`).then((response) => {
        this.setState({ blogEntry: response.body });
      });
    }
  }

  render() {
    return (
      this.state && this.state.blogEntry && 
      <div className="container">
        <article>
        <h1><a href={`/${this.state.blogEntry.slug}`} itemProp="url">{this.state.blogEntry.title}</a></h1>
        <div className="meta">
          <p>
            <time itemProp="dateCreated" dateTime={`${this.state.blogEntry.pub_date}`}>{ this.state.blogEntry.pretty_pub_date }</time>
             âˆ™ <a href={`https://github.com/eliakaris/blog/tree/master/posts/${this.state.blogEntry.slug}.md`}>History</a>
          </p>
        </div>
        <div dangerouslySetInnerHTML={{__html: this.state.blogEntry.html}} />
        </article>
      </div>
    );
  }
}

export default BlogEntry;