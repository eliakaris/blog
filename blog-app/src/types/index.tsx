
// src/types/index.tsx

import { BlogEntryData } from "./BlogEntry";

export interface HelloState {
  languageName: string;
  enthusiasmLevel: number;
}

export interface StoreState {
  blogEntries: BlogEntryData[];
  helloState: HelloState;
}
