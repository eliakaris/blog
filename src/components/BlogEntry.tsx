import * as React from 'react';
import { BlogEntryData } from '../../server/BlogEntry';
import './BlogEntry.css';
import 'highlight.js/styles/github.css';

export interface Props {
  blogEntry: BlogEntryData;
}

function BlogEntry({ blogEntry }: Props) {

  const postRoot = 'https://github.com/eliakaris/blog/tree/master/server/data/posts';
  return (
    blogEntry && 
    (
    <article>
      <h1><a href={`/blog/${blogEntry.slug}`} itemProp="url">{blogEntry.title}</a></h1>
      <div className="meta">
        <p>
          <time itemProp="dateCreated" dateTime={`${blogEntry.pub_date}`}>
            {blogEntry.pretty_pub_date}
          </time>
          &nbsp;âˆ™&nbsp;
          <a href={`${postRoot}/${blogEntry.slug}.md`}>
            History
          </a>
        </p>
      </div>
      <div dangerouslySetInnerHTML={{__html: blogEntry.html}} />
    </article>
    )
  );
}

export default BlogEntry;