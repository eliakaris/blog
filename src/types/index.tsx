// src/types/index.tsx
import { BlogEntryData, BlogListEntry } from '../../server/BlogEntry';

export interface StoreState {
  blogEntry: BlogEntryData;
  blogEntries: BlogListEntry[];
}
