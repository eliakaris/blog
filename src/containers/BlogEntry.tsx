import * as React from 'react';
import { Dispatch } from 'react-redux';
import BlogEntry, { Props } from '../components/BlogEntry';
import { StoreState } from '../types/index';
import { connect } from 'react-redux';
import * as actions from '../actions/index';
import { RequestBlogEntryAction } from '../actions/index';

interface BlogEntryContainerProps extends Props {
  match: {
    params: {
      slug: string;
    }
  };
  dispatch: Dispatch<RequestBlogEntryAction>;
}

class BlogEntryContainer extends React.Component<BlogEntryContainerProps> {

  componentDidMount() {
    this.props.dispatch(actions.getBlogEntry(this.props.match.params.slug));
  }

  render() {
    return (
      <BlogEntry {...this.props} />
    );
  }
}

export function mapStateToProps({ blogEntry }: StoreState) {
  return {
    blogEntry
  };
}

export default connect(mapStateToProps)(BlogEntryContainer);