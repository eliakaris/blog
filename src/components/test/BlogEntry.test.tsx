import * as React from 'react';
import BlogEntry from '../BlogEntry';
import * as ShallowRenderer from 'react-test-renderer/shallow';
import * as ReactSixteenAdapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';
import { BlogEntryData } from '../../types/BlogEntry';

configure({ adapter: new ReactSixteenAdapter() });

describe('<BlogEntry /> tests', () => {
  let renderer: any;

  beforeAll(() => {
    renderer = ShallowRenderer.createRenderer();
  });

  it('renders a snapshot of a blog entry', () => {
    const blogEntry: BlogEntryData = {
      html: "<div>hi there</div>",
      slug: "test-blog-entry",
      title: "test blog entry",
      summary: "summary for test blog entry",
      pretty_pub_date: "December 26th 2017",
      pub_date: "20171226"
    };

    const component = renderer.render(<BlogEntry blogEntry={blogEntry} />);
    expect(component).toMatchSnapshot();
  });
});