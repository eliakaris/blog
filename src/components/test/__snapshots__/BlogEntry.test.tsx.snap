// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogEntry /> tests renders a snapshot of a blog entry 1`] = `
<div
  className="row"
>
  <article>
    <h1>
      <a
        href="/blog/test-blog-entry"
        itemProp="url"
      >
        test blog entry
      </a>
    </h1>
    <div
      className="meta"
    >
      <p>
        <time
          dateTime="20171226"
          itemProp="dateCreated"
        >
          December 26th 2017
        </time>
         ∙ 
        <a
          href="https://github.com/eliakaris/blog/tree/master/server/data/posts/test-blog-entry.md"
        >
          History
        </a>
      </p>
    </div>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<div>hi there</div>",
        }
      }
    />
  </article>
</div>
`;
